src/Stratis.Features.Unity3dApi/NFTTransferIndexer.cs

```diff
@@ -188,14 +185,19 @@ public void ReindexAllContracts()
 
             int watchFromHeight = this.GetWatchFromHeight();
 
+            var updated = new List<NFTContractModel>();
+
             foreach (NFTContractModel contractModel in this.NFTContractCollection.FindAll().ToList())
             {
                 contractModel.OwnedIDsByAddress = new Dictionary<string, HashSet<long>>();
-                contractModel.LastUpdatedBlock = watchFromHeight;
 
-                this.NFTContractCollection.Upsert(contractModel);
+                updated.Add(contractModel);
             }
 
+            this.NFTContractCollection.Upsert(updated);
+
+            this.UpdateLastUpdatedBlock(0);
```

Comment:
I think that 0 should be the `watchFromHeight` ? (i.e. so that it starts at bliock 3 000 000 I think)?

Reply:
Good point, I will amend that